### ISSUE TITLE
Reconsider population of params

### ISSUE THREAD

**Issue by TejasQ** [Contiamo/rest-client#8]
[Reference](https://github.com/Contiamo/rest-client/blob/master/src/capabilities/resource.ts#L24)

I do not see the point in having `options` spread into `params.options` and then overwriting _any_ `method` property in `options` that the user might specify. This seems not just like a code smell, but a code _stench of death_ that I just inhaled while trying to work with this.

I'd like to repurpose this to place the spreading of `options` after pre-populating it with our recommended defaults.

**Comment by micha-f** [comment]
Did you accidentally overwrite the `method` property?

So you propose that the properties we define take precedence over anything the user provides execution time?

**Comment by micha-f** [comment]
Would probably make sense to "protect" our properties from being overwritten here as well:
https://github.com/Contiamo/rest-client/blob/master/src/collection.ts#L38
https://github.com/Contiamo/rest-client/blob/master/src/resource.ts#L57
https://github.com/Contiamo/rest-client/blob/master/src/client.ts#L119

**Comment by TejasQ** [comment]
@micha-f no! I _wanted_ to manually override it, but this library _fought me_, which feels awful as a consumer. I think our tools ought to assume developers using them _know what they're doing_ and don't necessarily _force them into using certain methods_ (in this case). What if I want `{ method: "PATCH" }` instead of `PUT` for this use-case mentioned in my initial comment? I can't. I'd have to write a whole nother method in my API blueprint definition of a given resource that is basically a copypasta of this code, but with `{ method: "PATCH" }` replaced. I do not think this is a good approach.

I propose that the properties we define _do not_ take precedence over what the user provides. As in with sales, _the customer is always right_ is the approach I feel we ought to take with this.

**Comment by micha-f** [comment]
@TejasQ got it. Sorry - that was the approach I was trying to take in this library in most places (see links I posted above). Feel free to make this more consistent and allow user-defined values to take precedence.
